core-tests-android/src/androidTest/java/com/powersync/AndroidDatabaseTest.kt

```diff
@@ -222,4 +222,10 @@ class AndroidDatabaseTest {
             // The exception messages differ slightly between drivers
             assertEquals(exception.message!!.contains("write a readonly database"), true)
         }
+
+    @Test
+    fun canCreateTemporaryTable() = runTest {
+        database.execute("PRAGMA temp_store = 1;") // Store temporary data as files
+        database.execute("CREATE TEMP TABLE my_tbl (content ANY) STRICT;")
```

Comment:
It's worth noting that this test passes even without the pragma, but I'm not sure why. My understanding is that SQLite should use temporary files here, but it apparently doesn't.

Reply:
I believe SQLite always keeps it in memory first, until it reaches a certain size threshold. I can't find any docs confirming this though.

Reply:
I tried inserting a 1MiB `randomblob` 128 times, but it still seems to pass without setting the temp directory.

Reply:
Another thing to check is the Android version it is being tested on - I remember this potentially only affecting newer Android versions. Although I can't find where the test emulator is specified here?

Reply:
I've just tested this manually with an SDK 35 emulator. An SDK-specific issue would surprise me though, we ship our own SQLite library along with the application. Or are you saying that some Android versions give apps their own `/tmp` that would be writable?

Reply:
Ok, I think I'm remembering anecdotes of temporary storage issues only being present on some Android versions, but I don't think we ever confirmed that.

Either way it doesn't quite explain how these tests worked, unless the emulator uses different filesystem permission structures from actual devices?

Reply:
> unless the emulator uses different filesystem permission structures from actual devices?

That sounds unlikely, the emulator is a VM running a real Android system.

> I think I'm remembering anecdotes of temporary storage issues only being present on some Android versions

I've asked the user reporting this to share the Android version they're using. Maybe it's not a problem on newer Android versions and that's why we didn't catch this.